### PR TITLE
Some janitoring

### PR DESCRIPTION
**Pull request by akva2**
Quell some warnings.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/wells/MultisegmentWell_impl.hpp [hunk 1]
```diff
@@ -887,6 +887,7 @@ namespace Opm
         auto obtain = [this](const Eval& value)
                       {
                           if constexpr (std::is_same_v<Value, Scalar>) {
+                              (void)this; // suppress clang warning
```
Comment (bska) [inline opm/simulators/wells/MultisegmentWell_impl.hpp:890]: Would you mind writing `static_cast<>()` please?  That stands out more clearly when reading the code.

**Review by bska — COMMENTED** [review]
Much appreciated.  I'd only like to see us use C++ casts instead of C-style casts.  Both because they are more obvious when reading the code and because they're more well behaved.

File: opm/simulators/wells/MultisegmentWell_impl.hpp (hunk 1 above)
Comment (akva2) [reply opm/simulators/wells/MultisegmentWell_impl.hpp]: sure.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Perfect!  Thanks a lot for the updates.  Please merge when green.

**Pull request merged**